motors/tacho_motor_class.c

```
@@ -942,12 +1179,28 @@ int register_tacho_motor(struct tacho_motor_device *tm, struct device *parent)
 	
 	tm->dev.release = tacho_motor_release;
 	tm->dev.parent = parent;
+
+	switch (tm->ops->get_motion_type(tm->context)) {
+	case TM_MOTION_LINEAR:
+		tacho_motor_class.dev_groups = tacho_motor_linear_groups;
+		dev_set_name(&tm->dev, "linear%d", tacho_motor_class_id++);
+		break;
+	case TM_MOTION_ROTATION:
+	default:
```

There's times when `default` should return an error so that you know you have an unhandled case - this is one of them